neqo-crypto/src/agent.rs

```diff
@@ -50,6 +50,110 @@ use crate::{
     time::{Time, TimeHolder},
 };
 
+/// Private trait for Certificate Compression implementation
+/// # Safety
+///
+/// Use `SafeCertCompression` to implement an encoder/decoder instead.
+unsafe trait UnsafeCertCompression {
+    unsafe extern "C" fn decode_callback(
+        input: *const ssl::SECItem,
+        output: *mut ::std::os::raw::c_uchar,
+        output_len: usize,
+        used_len: *mut usize,
+    ) -> ssl::SECStatus;
+
+    unsafe extern "C" fn encode_callback(
+        input: *const ssl::SECItem,
+        output: *mut ssl::SECItem,
+    ) -> ssl::SECStatus;
+}
+
+/// The trait is used to represent a certificate compression data structure
+/// Used in order to enable Certificate Compression extension during TLS connection
+pub trait SafeCertificateCompression {
+    /// Certificate Compression identifier as in RFC8879
+    const ID: u16;
+    /// Certification Compression name (used only for logging/debugging)
+    const NAME: &str;
+    /// Certificate Compression could be used to encode and decode a certificate
+    /// though the encoding is not frequently used
+    /// Enable decoding field is used to signal to the implementation
+    /// to use the encoding as well
+    const ENABLE_ENCODING: bool = false;
+
+    /// Certificate Compression encoding function
+    /// If the implementation is not provided, we only copy the data
+    /// NB: If `ENABLE_ENCODING` is not set, the function pointer provided to NSS will be null
+    #[expect(clippy::must_use_candidate, reason = "Encoders are optional")]
```

Comment:
```suggestion
    #[must_use]
```

Anyone calling this function should use the returned value.  That's what `must_use` is for.  This interface allows it to be left `unimplemented!()` if `ENABLE_ENCODING` is false, that's all.